src/Vixen.Modules/Editor/TimedSequenceEditor/EffectModelCandidate.cs

```diff
@@ -12,19 +12,21 @@ namespace VixenModules.Editor.TimedSequenceEditor
 	[Serializable]
 	public class EffectModelCandidate
 	{
-		private readonly Type _moduleDataClass;
-		private readonly MemoryStream _effectData;
+		private readonly string _moduleDataClass;
+		private readonly byte[] _effectData;
 
 		public EffectModelCandidate(IEffectModuleInstance effect)
 		{
 			LayerId = Guid.Empty;
-			_moduleDataClass = effect.Descriptor.ModuleDataClass;
-			DataContractSerializer ds = new DataContractSerializer(_moduleDataClass);
+			_moduleDataClass = effect.Descriptor.ModuleDataClass.AssemblyQualifiedName;
+			DataContractSerializer ds = new DataContractSerializer(Type.GetType(_moduleDataClass));
 

```

Comment:
This code change while functional, introduces potential unchecked null object exceptions. The AssemblyQualifiedName returns a nullable string?, meaning it could be null. That is then passed into the GetType method that the string parameter is not nullable. So this code has the potential to throw a null exception if the AssemblyQualifiedName is null. Could it be null? Probably not, but the method signature does not guarantee it, so the potential is there. Similarly the DataContractSerializer takes a not null Type, but GetType can return null since it is defined as nullable return field of Type?. So another potential null exception. The solution here is to add null checking to validate and control the flow. This could also mean upstream concerns since this constructor may not complete in a successful way. We have been used to writing this type of code for a long time assuming things like this should not ever be null, but signatures of the operations here do not guarantee that.